sockets.h

```diff
@@ -39,17 +40,20 @@ bool readSocket(WiFiClient& client, uint8_t dataType) {
         //printSocketBuff();
         
         res = true;
+        delete[] tmp;
```

Comment:
why do we delete this here but not out of condition ?

Reply:
Because we allocate memory inside condition. 
If this condition occurs multiple times, we`ll have memory leaks(memory will be allocated multiple times, but freed once).
Need to be tested. If this condition occurs once - then delete outside is ok.

Reply:
and what if we allocate memory in other branches of this condition?
should we call delete[] each time?

my guess is following. we create pointer. we allocate memory in the way accordingly to condition. we delete[] pointer without taking care about how was this memory allocated (or it was not at all)

is this vision correct?

check pls

Reply:
This can cause memory leaks. The easiest way is to use smart pointer class - unique_ptr<T>.

But I do not sure about this. 
Maybe it is better to rewrite this code so, that there will be no manually memory allocation(no new and delete).